executor/src/witgen/generator.rs

```diff
@@ -485,11 +504,17 @@ where
                     assert!(!progress);
                 }
                 for (id, c) in constraints.constraints {
+                    if first {
+                        log::trace!("    Processing: {}", source_name());
+                        first = false;
+                    }
                     match c {
                         Constraint::Assignment(value) => {
+                            log::trace!("      => {id} = {value}");
                             self.next[id.poly_id() as usize] = Some(value);
                         }
                         Constraint::RangeConstraint(cons) => {
+                            log::trace!("      => Adding range constraint for {id}: {cons}");
```

Comment:
Maybe we should also print the new range constraint that is the conjunction of the old one with this one. The result might be less specific than we expect.